### PR TITLE
new package: softether-vpn

### LINKED ISSUE #2509: Package Request: Softether Client
SoftEther VPN ("SoftEther" means "Software Ethernet") is one of the world's most powerful and easy-to-use multi-protocol VPN software.

### LINKED ISSUE #2883: Package Request: Softether Server
SoftEther VPN ("SoftEther" means "Software Ethernet") is one of the world's most powerful and easy-to-use multi-protocol VPN software.

SoftEther provides SSTP, L2TP, OpenVPN and their own specific SoftEther protocol.

SoftEther source code for ARM EABI 32-bit Linux is provided here:
https://www.softether-download.com/en.aspx?product=softether

Have a look at build notes:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md

### PR DESCRIPTION
**Pull request by xtkoba**
Closes #2883, closes #2509.

### CODE REVIEW

File: packages/softether-vpn/build.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+TERMUX_PKG_HOMEPAGE=https://www.softether.org/
+TERMUX_PKG_DESCRIPTION="An open-source cross-platform multi-protocol VPN program"
+TERMUX_PKG_LICENSE="Apache-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=(5.02.5180)
+TERMUX_PKG_VERSION+=(1.0.18)
+TERMUX_PKG_SRCURL=(https://github.com/SoftEtherVPN/SoftEtherVPN/releases/download/${TERMUX_PKG_VERSION}/SoftEtherVPN-${TERMUX_PKG_VERSION}.tar.xz
+                   https://github.com/jedisct1/libsodium/archive/${TERMUX_PKG_VERSION[1]}-RELEASE.tar.gz)
```
Comment (Grimler91) [inline packages/softether-vpn/build.sh:8]: Is libsodium sources used in compilation of softethervpn, or only for hostbuild step? Can't libsodium be [installed to the docker image](https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh) instead?

File: packages/softether-vpn/build.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+TERMUX_PKG_HOMEPAGE=https://www.softether.org/
+TERMUX_PKG_DESCRIPTION="An open-source cross-platform multi-protocol VPN program"
```
Comment (Grimler91) [inline packages/softether-vpn/build.sh:2]: Is this package used for setting up an softether vpn server, or is it a client program for connecting to a server?
Does it work without root?

File: packages/softether-vpn/build.sh (hunk 1 above)
Comment (xtkoba) [reply packages/softether-vpn/build.sh:8]: Only for host build. Can be replaced by a binary package in docker.

Strictly, libsodium for host build should not be required at all. We only need to build the `hamcorebuilder` binary for running on build host, which does not depend on libsodium. But there seems to be no trivial way to build it separately.

File: packages/softether-vpn/build.sh (hunk 2 above)
Comment (xtkoba) [reply packages/softether-vpn/build.sh:2]: Contains both client and server. Some part requires root and other parts do not.

Again, the problem is that there is no trivial way to separate client and server builds.

**Review by xeffyr — CHANGES_REQUESTED** [review]
For now it installs files for systemd:
```
/data/data/com.termux/files/usr/lib/systemd/*
```

Remove them. Termux doesn't have or supports systemd init.

**Comment by xtkoba** [comment]
`build.sh` already has the following line:

```bash
TERMUX_PKG_RM_AFTER_INSTALL="lib/systemd"
```

And actually the package does not have those files. Is this insufficient?

**Comment by xeffyr** [comment]
Ah, sorry. This should be sufficient. Just there were no artifacts to download and I reviewed it through build logs.

File: packages/softether-vpn/build.sh (hunk 2 above)
Comment (DRSDavidSoft) [reply packages/softether-vpn/build.sh:2]: @xtkoba As a user, I personally don't mind that. I appreciate it if I could run the SoftEther server as well as the client from the same package

**Pull request merged**